### DISCUSSION TITLE
With 500 servers using Redisson, who does the cleanup work (evicting, updating time-outs, etc)?

### DISCUSSION THREAD

**Discussion by contentfree** [redisson/redisson#5405] [Q&A]
We're running into some very strange caching behavior (stop getting correct values for one minute). While investigating, this question came up: When there are more than one server using Redisson, how does the management work get done and who does it?

**Comment by mrniko** [comment]
Which management work do you mean?

**Comment by contentfree** [comment]
The work of evicting keys from the regions, tracking idle timeouts, etc.
That isn't done by Redis itself so presumably it's done by Redisson. Which
of my hundreds of servers using Redisson to talk to Redis does that work?

**Comment by contentfree** [comment]
Ha… I hate it when I'm looking for an answer to a question on the internet and find my _own_ question from year(s) ago. 

@mrniko Can you assist with an answer here? If I have 1000 consumers using redisson to talk to redis, do all of them run the housekeeping task that cleans up expired keys? 

If so, is it possible to _not_ do that and tasking a subset of those workers with the housekeeping? 

(I ask because our redis node – an m7g.xlarge – is often at 92% for the engine CPU utilization in AWS Elasticache … and I need to figure out if there's something I should do before scaling horizontally)

**Comment by contentfree** [comment]
Any thoughts here, @mrniko ?

**Reply by mrniko** [reply]
Cleanup process is used for RMapCache object. You can RMapCacheNative if you have Redis 7.4.0+ version or Redisson PRO

**Comment by contentfree** [comment]
Does each Redisson-using client run their own cleanup process? Or is there
some internal cleanup process in Redis? And how do we tell Redisson to use
RMapCacheNative?
